CMakePresets.json

```diff
@@ -22,15 +22,19 @@
                 "microsoft.com/VisualStudioSettings/CMake/1.0": {
                     "hostOS": [ "Windows" ]
                 }
+            },
+            "condition": {
+              "type": "equals",
+              "lhs": "${hostSystemName}",
+              "rhs": "Windows"
             }
         },
         {
             "name": "vc17-x64-windows",
             "hidden": true,
             "inherits": "x64-windows",
             "cacheVariables": {
-                "CMAKE_CXX_COMPILER": "cl",
-                "CMAKE_C_COMPILER": "cl"
+                "CMAKE_CXX_COMPILER": "cl"
```

Comment:
I explicitly added CMAKE_C_COMPILER otherwise CMake sometimes says "there are no compilers that can make a program"

Reply:
ok, good to know. will revert.
i don't get your error message, i get a CMake warning message: "unused directive: CMAKE_C_COMPILER" instead.